src/main/java/com/code_intelligence/jazzer/Jazzer.java

```diff
@@ -249,21 +257,30 @@ private static String prepareArgv0(Map<String, String> additionalEnvironment) th
     return launcher.toAbsolutePath().toString();
   }
 
-  private static Path javaBinary() {
+  private static String javaBinary() {
     String javaBinaryName;
     if (isAndroid()) {
-      javaBinaryName = "dalvikvm";
+      javaBinaryName = "exec";
+      return javaBinaryName;
     } else if (isPosix()) {
       javaBinaryName = "java";
     } else {
       javaBinaryName = "java.exe";
     }
 
-    return Paths.get(System.getProperty("java.home"), "bin", javaBinaryName);
+    Path path = Paths.get(System.getProperty("java.home"), "bin", javaBinaryName);
+    return path.toString();
   }
 
   private static Stream<String> javaBinaryArgs() {
-    Stream<String> stream = Stream.of("-cp", System.getProperty("java.class.path"),
+    Stream<String> stream;
+    if (isAndroid()) {
+      stream = Stream.of("app_process", "-Djdk.attach.allowAttachSelf=true", "/system/bin",
+          Jazzer.class.getName());
+      // ManagementFactory wont work with Android
+      return stream;
```

Comment:
```suggestion
      return Stream.of("app_process", "-Djdk.attach.allowAttachSelf=true", "/system/bin",
          Jazzer.class.getName());
```

No need to use the local variable if you return early.